pkg/policy/correlation/correlation.go

```diff
@@ -64,34 +72,39 @@ func CorrelatePolicy(endpointGetter getters.EndpointGetter, f *flowpb.Flow) {
 		return
 	}
 
-	allowedBy := toProto(derivedFrom, rev)
-	switch direction {
-	case trafficdirection.Egress:
-		f.EgressAllowedBy = allowedBy
-	case trafficdirection.Ingress:
-		f.IngressAllowedBy = allowedBy
+	rules := toProto(derivedFrom, rev)
+	switch {
+	case direction == trafficdirection.Egress && allowed:
+		f.EgressAllowedBy = rules
+	case direction == trafficdirection.Egress && denied:
+		f.EgressDeniedBy = rules
+	case direction == trafficdirection.Ingress && allowed:
+		f.IngressAllowedBy = rules
+	case direction == trafficdirection.Ingress && denied:
+		f.IngressDeniedBy = rules
```

Comment:
maybe add `default: log.Debug("unexpected state")` or something like that so if there are ever issues we can try to debug trace it

Reply:
Given the prevalence of policy verdict events, I'm a bit worried that this would create CI noise similar as other Hubble debug messages we added for similar purposes: https://github.com/cilium/cilium/issues/15283

Given that a "unexpected state" will manifest itself in no field being populated, the missing case will be detectable. Thus I'll leave as is for now.